src/main/java/tel/schich/javacan/BcmMessage.java

```diff
@@ -0,0 +1,363 @@
+     * interval. When {@link BcmFlag#STARTTIMER} is set, the timeout detection is activated directly -
+     * even without a former CAN frame reception.</li>
+     * </ul>
+     *
+     * @return the duration or {@code null} if it is not set
+     */
+    public Duration getInterval1() {
+        long sec = getPlatformLong(base + OFFSET_IVAL1_TV_SEC);
+        long usec = getPlatformLong(base + OFFSET_IVAL1_TV_USEC);
+        return (sec + usec) != 0 ? Duration.ofSeconds(sec).plusMillis(usec) : null;
+    }
+
+    /**
+     * The {@code interval2} has different meanings depending on the {@link #getOpcode()} of the
+     * message:
+     * <ul>
+     * <li><b>When used with {@link BcmOpcode#TX_SETUP}:</b></br>
+     * see {@link #getInterval1()}</li>
+     * <li><b>When used with {@link BcmOpcode#RX_SETUP}:</b></br>
+     * Throttle the received message rate down to the value of {@code interval2}. This is useful to
+     * reduce messages for the application when the signal inside the CAN frame is stateless as state
+     * changes within the {@code interval2} duration may get lost.</li>
+     * </ul>
+     *
+     * @return the duration or {@code null} if it is not set
+     */
+    public Duration getInterval2() {
+        long sec = getPlatformLong(base + OFFSET_IVAL2_TV_SEC);
+        long usec = getPlatformLong(base + OFFSET_IVAL2_TV_USEC);
+        return (sec + usec) != 0 ? Duration.ofSeconds(sec).plusMillis(usec) : null;
+    }
+
+    /**
+     * Returns the CAN ID of this message.
+     */
+    public int getCanId() {
+        return buffer.getInt(base + OFFSET_CAN_ID);
+    }
+
+    /**
+     * Returns the number of frames in this message.
+     */
+    public int getFrameCount() {
+        return buffer.getInt(base + OFFSET_NFRAMES);
+    }
+
+    /**
+     * Returns a single frame of this message.
+     *
+     * @param index of the frame; ({@code 0 <= index < frameCount})
+     * @return the frame
+     * @throws IllegalArgumentException if the message buffer contains no frame for that index
+     */
+    public CanFrame getFrame(int index) {
+        ByteBuffer frameBuffer = buffer.duplicate();
+        int frameLength = frameLength(getFlags());
+        frameBuffer
+                .position(base + OFFSET_FRAMES + index * frameLength)
+                .limit(frameBuffer.position() + frameLength);
+        return CanFrame.create(frameBuffer);
+    }
+
+    /**
+     * Returns all frames of this message.
+     *
+     * @return all frames of this message
+     */
+    public List<CanFrame> getFrames() {
+        int nFrames = getFrameCount();
+        if (nFrames == 0) {
+            return Collections.emptyList();
+        }
+        int frameLength = frameLength(getFlags());
+        List<CanFrame> frames = new ArrayList<>(nFrames);
+        for (int i = 0; i < nFrames; i++) {
+            ByteBuffer frameBuffer = buffer.duplicate();
```

Comment:
this code is identical to the code in `getFrame(int)`. Mabye extract a private method `getFrame(int index, int length)`

Reply:
okay